haystack/nodes/prompt/providers.py

```diff
@@ -662,3 +715,97 @@ def url(self) -> str:
+            raise RuntimeError(f"model_name_or_path {model_name_or_path} must be a valid Cohere model name. ")
+        self.api_key = api_key
+        self.max_length = max_length or 128
+
+        # See https://docs.cohere.ai/reference/generate for more details on the parameters
+        self.model_input_kwargs = {
+            key: kwargs[key]
+            for key in ["top_k", "top_p", "temperature", "stop_sequences", "max_tokens"]
+            if key in kwargs
+        }
+        # Cohere API has a max token limit of 2048
+        # Cohere doesn't provide tokenizer Python lib; they mention use BPE tokenizer (i.e GPT2, RoBERTa, XLM, etc.)
+        # See https://docs.cohere.ai/reference/tokenize
+        self.token_enforcer = HFTokenLimitEnforcer(
+            tokenizer=AutoTokenizer.from_pretrained("gpt2"), answer_max_length=self.max_length, model_max_length=2048
+        )
+        self.client = HTTPClient(
+            method="POST",
+            url="https://api.cohere.ai/v1/generate",
+            headers=self.headers,
+            response_handler=lambda r: [o["text"] for o in r.json()["generations"]],
+            default_error_class=CohereError,
+            error_codes_map={401: CohereUnauthorizedError},
+        )
+
+    @property
+    def headers(self) -> Dict[str, str]:
+        return {"Authorization": f"Bearer {self.api_key}", "Content-Type": "application/json"}
+
+    def invoke(self, *args, **kwargs):
+        """
+        Invokes a prompt on the model. It takes in a prompt and returns a list of responses using a REST invocation.
+
+        :return: The responses are being returned.
+        """
+        prompt = kwargs.get("prompt")
+        if not prompt:
+            raise ValueError(
+                f"No prompt provided. Model {self.model_name_or_path} requires prompt."
+                f"Make sure to provide prompt in kwargs."
+            )
+        # we standardize on keyword stop_words but Cohere uses stop_sequences
+        stop_words = kwargs.pop("stop_words", None)
+        kwargs["stop_sequences"] = stop_words
+
+        # Cohere command can't generate more than one response
+        kwargs.pop("top_k", None)  # type: ignore
+        kwargs_with_defaults = self.model_input_kwargs
+        kwargs_with_defaults.update(kwargs)
+        payload = {
+            "prompt": prompt,
+            "model": self.model_name_or_path,
+            "stop_sequences": kwargs_with_defaults.get("stop_sequences", None),
+            "max_tokens": kwargs_with_defaults.get("max_length", self.max_length),
+            "temperature": kwargs_with_defaults.get("temperature", 0.9),
+            "top_p": kwargs_with_defaults.get("top_p", 0.75),
+        }
+
+        sentences = self.client.request(json=payload)
+        # Cohere stops generation on stop words, but we also want to remove them from the output
+        return remove_words(sentences, stop_words) if stop_words else sentences
```

Comment:
Nitpick: you don't need the if. If there are no stopwords, remove_words is basically no-op
```suggestion
        return remove_words(sentences, stop_words)
```